src/services/item/plugins/importExport/itemExportRequest.service.ts

```diff
@@ -0,0 +1,172 @@
+
+const EXPORT_ITEM_EXPIRATION_DAYS = 7;
+const EXPORT_ITEM_EXPIRATION = 3600 * 24 * EXPORT_ITEM_EXPIRATION_DAYS; // max value: one week
+
+export const ItemRequestExportType = {
+  Raw: 'raw',
+  Graasp: 'graasp',
+} as const;
+type ItemRequestExportTypeOptions = UnionOfConst<typeof ItemRequestExportType>;
+
+@singleton()
+export class ItemExportRequestService {
+  private readonly fileService: FileService;
+  private readonly itemRequestExportRepository: ItemRequestExportRepository;
+  private readonly mailerService: MailerService;
+  private readonly memberService: MemberService;
+  private readonly logger: BaseLogger;
+
+  constructor(
+    fileService: FileService,
+    itemRequestExportRepository: ItemRequestExportRepository,
+    mailerService: MailerService,
+    memberService: MemberService,
+    logger: BaseLogger,
+  ) {
+    this.fileService = fileService;
+    this.itemRequestExportRepository = itemRequestExportRepository;
+    this.mailerService = mailerService;
+    this.memberService = memberService;
+    this.logger = logger;
+  }
+
+  /**
+   * path where an export request is saved on the file storage
+   * @param requestId
+   * @returns
+   */
+  private buildExportPath(requestId: string) {
+    return `item-export/${requestId}`;
+  }
+
+  /**
+   * Upload archive and send email with the download link
+   * @param dbConnection
+   * @param actor
+   * @param item
+   * @param archive zip content of the item to be uploaded
+   */
+  async uploadAndSendDownloadLink(
+    dbConnection: DBConnection,
+    actor: MinimalAccount,
+    item: ItemRaw,
+    archive: ZipFile,
+    type: ItemRequestExportTypeOptions,
+  ) {
+    // upload zip
+    const filepath = await this.uploadZip(dbConnection, actor, item, archive, type);
+
+    // send email
+    const memberWithEmail = (await this.memberService.get(dbConnection, actor.id)).toMemberInfo();
+    await this.sendExportRawLinkInMail(memberWithEmail.email, memberWithEmail.lang, filepath, item);
+  }
+
+  /**
+   * Upload zip on file storage service
+   * @param dbConnection
+   * @param actor
+   * @param item
+   * @param archive
+   * @returns file path where the file is saved on the server
+   */
+  private async uploadZip(
+    dbConnection,
+    actor,
+    item,
+    archive: ZipFile,
+    type: ItemRequestExportTypeOptions,
+  ) {
+    const request = await this.itemRequestExportRepository.addOne(dbConnection, {
+      memberId: actor.id,
+      itemId: item.id,
+      type,
+    });
+
+    const filepath = this.buildExportPath(request.id);
+    const tmpFolder = path.join(TMP_FOLDER, 'item-export');
+    // make sure tmp folder exists
+    fs.mkdirSync(tmpFolder, { recursive: true });
+    const tmpFilepath = path.join(tmpFolder, request.id);
+
+    try {
+      // save archive content to stream
+      await pipeline(archive.outputStream, fs.createWriteStream(tmpFilepath));
+      const readFile = fs.createReadStream(tmpFilepath);
+
```

Comment:
Is it necessary to have the temporary file on disk ? Would it be possible to pipe the stream directly to s3/local ?